### PR TITLE
Select more specific method as per return type

### PR DESCRIPTION
**Pull request by JasonFengJ9**
Select more specific method as per return type

Return type `S` is more specific than another return type `R` if `R.class.isAssignableFrom(java.lang.Class<?>)(S.class)`;
Select a more specific return type instead of a return type with larger
class depth;
Removed native `j.l.Class.getClassDepth()`;
Added tests to verify new behaviours.

closes: #2162 

Reviewer: @DanHeidinga 

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

### CODE REVIEW

File: test/functional/Java8andUp/src_80/org/openj9/test/java/lang/Test_Class.java [hunk 1]
```diff
@@ -442,7 +442,71 @@ public void test_getMethods_subtest5() {
 			}
 		}
 	}
-
+	@Test
+	public void test_getMethods_subtest6() {
+		try {
+			TestSpecificMethodsCL	tsmcl = new TestSpecificMethodsCL();
```
Comment (DanHeidinga) [inline test/functional/Java8andUp/src_80/org/openj9/test/java/lang/Test_Class.java]: nitpick: spacing

File: test/functional/Java8andUp/src_80/org/openj9/test/java/lang/Test_Class.java [hunk 2]
```diff
@@ -442,7 +442,71 @@ public void test_getMethods_subtest5() {
...
+			AssertJUnit.assertTrue("Expected method return type: " + java.util.List.class + " but got: " + specificMethod.getReturnType(), 
+				java.util.List.class.equals(specificMethod.getReturnType()));
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+	}
+	@Test
+	public void test_getMethods_subtest7() {
+		try {
+			TestSpecificMethodsCL	tsmcl = new TestSpecificMethodsCL();
```
Comment (DanHeidinga) [inline test/functional/Java8andUp/src_80/org/openj9/test/java/lang/Test_Class.java]: nitpick spacing

File: test/functional/Java8andUp/src_80/org/openj9/test/java/lang/Test_Class.java (hunk 1 above)
Comment (JasonFengJ9) [reply test/functional/Java8andUp/src_80/org/openj9/test/java/lang/Test_Class.java]: Applied `Eclipse format`, fixed.

File: test/functional/Java8andUp/src_80/org/openj9/test/java/lang/Test_Class.java (hunk 2 above)
Comment (JasonFengJ9) [reply test/functional/Java8andUp/src_80/org/openj9/test/java/lang/Test_Class.java]: Fixed.

File: test/functional/Java8andUp/src_80/org/openj9/test/java/lang/Test_Class.java [hunk 3]
```diff
@@ -442,7 +442,71 @@ public void test_getMethods_subtest5() {
...
+			Class<?> clz = tsmcl.loadClass("TestSpecificMethods");
+			Method specificMethod = clz.getMethod("foo");
+			AssertJUnit.assertTrue("Expected method return type: " + java.util.List.class + " but got: " + specificMethod.getReturnType(), 
+				java.util.List.class.equals(specificMethod.getReturnType()));
+		} catch (Exception e) {
```
[the remark this answers is not in the document]
Comment (JasonFengJ9) [reply test/functional/Java8andUp/src_80/org/openj9/test/java/lang/Test_Class.java]: Yeah, thanks for the reminder, fixed now.

File: test/functional/Java8andUp/src_80/org/openj9/test/java/lang/Test_Class.java [hunk 4]
```diff
@@ -442,7 +442,71 @@ public void test_getMethods_subtest5() {
...
-
+	@Test
+	public void test_getMethods_subtest6() {
+		try {
+			TestSpecificMethodsCL	tsmcl = new TestSpecificMethodsCL();
+			Class<?> clz = tsmcl.loadClass("TestSpecificMethods");
+			Method specificMethod = clz.getDeclaredMethod("foo");
+			AssertJUnit.assertTrue("Expected method return type: " + java.util.List.class + " but got: " + specificMethod.getReturnType(), 
+				java.util.List.class.equals(specificMethod.getReturnType()));
+		} catch (Exception e) {
```
[the remark this answers is not in the document]
Comment (JasonFengJ9) [reply test/functional/Java8andUp/src_80/org/openj9/test/java/lang/Test_Class.java]: Fixed.

File: test/functional/Java8andUp/src_80/org/openj9/test/java/lang/Test_Class.java [hunk 5]
```diff
@@ -442,7 +442,71 @@ public void test_getMethods_subtest5() {
...
 		}
 	}
-
+	@Test
+	public void test_getMethods_subtest6() {
+		try {
+			TestSpecificMethodsCL	tsmcl = new TestSpecificMethodsCL();
+			Class<?> clz = tsmcl.loadClass("TestSpecificMethods");
+			Method specificMethod = clz.getDeclaredMethod("foo");
+			AssertJUnit.assertTrue("Expected method return type: " + java.util.List.class + " but got: " + specificMethod.getReturnType(),
```
[the remark this answers is not in the document]
Comment (JasonFengJ9) [reply test/functional/Java8andUp/src_80/org/openj9/test/java/lang/Test_Class.java]: Agreed, replaced `Assert` with `AssertJUnit`, `assertTrue` with `assertEquals`.

**Comment by DanHeidinga** [comment]
Jenkins test sanity plinux jdk8

**Comment by DanHeidinga** [comment]
Jenkins test sanity plinux jdk8

**Pull request merged**